src/java.base/share/classes/sun/security/provider/HSS.java

```diff
@@ -0,0 +1,837 @@
+                result &= LMSUtils.lmsVerify(lmsPubKey, sig.siglist[i], keyArr);
+                lmsPubKey = sig.pubList[i];
+            }
+
+            result &= LMSUtils.lmsVerify(
+                    lmsPubKey, sig.siglist[sig.Nspk], messageStream.toByteArray());
+        } finally {
+            messageStream.reset();
+        }
+        return result;
+    }
+    @Override
+    protected void engineSetParameter(AlgorithmParameterSpec params)
+            throws InvalidAlgorithmParameterException {
+        if (params != null) {
+            throw new InvalidAlgorithmParameterException("No parameters accepted");
+        }
+    }
+
+    @Override
+    protected AlgorithmParameters engineGetParameters() {
+        return null;
+    }
+
+    static class LMSPublicKey {
+        final int type;
+        final int otsType;
+        final LMSParams lmsParams;
+        private final byte[] I;
+        private final byte[] T1;
+
+        LMSPublicKey(byte[] keyArray, int offset, boolean checkExactLength)
+                throws InvalidKeyException {
+            int inLen = keyArray.length - offset;
+            if (inLen < 8) {
+                throw new InvalidKeyException("LMS public key is too short");
+            }
+            type = LMSUtils.fourBytesToInt(keyArray, offset);
+            otsType = LMSUtils.fourBytesToInt(keyArray, offset + 4);
+            LMOTSParams lmotsParams;
+
+            try {
+                lmsParams = LMSParams.of(type);
+                lmotsParams = LMOTSParams.of(otsType);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidKeyException(e.getMessage());
+            }
+
+            int m = lmsParams.m;
+            if ((inLen < (24 + m)) || (checkExactLength && (inLen != (24 + m))) ||
+                    !lmsParams.hasSameHash(lmotsParams)) {
+                throw new InvalidKeyException("Wrong LMS public Key length");
+            }
+
+            I = Arrays.copyOfRange(keyArray, offset + 8, offset + 8 + 16);
+            T1 = Arrays.copyOfRange(keyArray, offset + 24, offset + 24 + m);
+        }
+
+        void getI(byte[] arr, int pos) {
+            System.arraycopy(I, 0, arr, pos, 16);
+        }
+
+        boolean isT1(byte[] arr, int pos) {
+            int m = lmsParams.m;
+            int diff = 0;
+            for (int i = 0; i < m; i++) {
+                diff |= (T1[i] ^ arr[pos + i]);
+            }
+            return (diff == 0);
+        }
+
+        byte[] keyArray() {
+            byte[] result = new byte[keyArrayLength()];
+            LMSUtils.intToFourBytes(type, result, 0);
+            LMSUtils.intToFourBytes(otsType, result, 4);
+            System.arraycopy(I, 0, result, 8, 16);
+            System.arraycopy(T1, 0, result, 24, lmsParams.m);
+            return result;
+        }
+
+        int keyArrayLength() {
+            return 24 + lmsParams.m;
+        }
+    }
+
+    static class LMSUtils {
+        final static int LMS_RESERVED = 0;
```

Comment:
use blessed modifiers order
```suggestion
        static final int LMS_RESERVED = 0;
```